modules/openapi-generator/src/main/resources/typescript-node/api-single.mustache

```diff
@@ -197,33 +197,35 @@ export class {{classname}} {
 {{/bodyParam}}
         };
 
+        let authenticationPromise = Promise.resolve();
```

Comment:
what is the use of this promise?

Reply:
To unconditionally setup the serial promise chain below.  It avoids `=== null` style checks.

An alternative I considered was executing the promises concurrently, storing them in an array, and using `Promise.all`.  That would change the semantics the the current synchronous and serial behavior.

Reply:
ah, ok, now I get it, the shorthand `return this.authentications.default.applyToRequest(localVarRequestOptions)` is the relevant part here.